geomstats/geometry/discrete_curves.py

```diff
@@ -1936,6 +1987,265 @@ def horizontal_path(t):
+            where gamma restricted to [x_min,xmax] is the linear
+            interpolation between y_min et y_max.
+            gamma(x_min) = y_min and gamma(x_max) = y_max
+            """
+            # Gamma's parameters
+            gamma_slope = (y_max - y_min) / (x_max - x_min)
+            gamma_constant = (y_min - x_min * gamma_slope)  # gamma_constant * n
+
+            # List of n*x where gamma(x) crosses a square
+            list_l = [x_i for x_i in range(x_min, x_max + 1)]
+            list_k = [(k - y_min) / gamma_slope + x_min
+                      for k in range(y_min, y_max + 1)]
+
+            lower_bound = x_min
+            i = 1
+            j = 1
+
+            """
+            At each step we compute value_1 and value_2 which are related to
+            the value of the integral restricted to [lower_bound,upper_bound]:
+            value = gamma_slope^(1/2) ( value_1*gamma_slope + value_2) / n
+            We can compute value_1 and value_2 in closed form, we have introduced
+            them to make the least amount of operation : the multiplications by
+            gamma_slope, gamma_slope^(1/2) and the division by n is done at the
+            very end instead of doing it at every interval.
+            """
+
+            value_1 = 0.
+            value_2 = 0.
+            value_3 = 0.
+            while i < x_max - x_min + 1 and j < y_max - y_min + 1:
+
+                # Integral's parameters
+                upper_bound = min(list_l[i], list_k[j])
+
+                q_1_slope = q_1[x_min + i] - q_1[x_min + i - 1]  # q_1_slope * n
+                q_2_slope = q_2[y_min + j] - q_2[y_min + j - 1]  # q_2_slope * n
+                q_1_constant = q_1[x_min + i - 1] - (x_min + i - 1) * q_1_slope
+                q_2_constant = q_2[y_min + j - 1] - (y_min + j - 1) * q_2_slope
+
+                # Compute the value_1, value_2 and value_3
+
+                vector_1 = q_2_slope * gamma_constant + q_2_constant
+
+                value_1 += np.dot(q_1_slope, q_2_slope) * \
+                    (math.pow(upper_bound, 3) - math.pow(lower_bound, 3))
+                value_2 += (np.dot(q_1_slope, vector_1) +
+                            gamma_slope * np.dot(q_2_slope, q_1_constant)) * \
+                    (math.pow(upper_bound, 2) - math.pow(lower_bound, 2))
+
+                value_3 += np.dot(q_1_constant, vector_1) * (upper_bound - lower_bound)
+
+                if list_l[i] == list_k[j]:
+                    i += 1
+                    j += 1
+                elif list_l[i] < list_k[j]:
+                    i += 1
+                else :
+                    j += 1
+                lower_bound = upper_bound
+
+            value = math.pow(gamma_slope, 1 / 2) * \
+                ((gamma_slope * value_1) / 3 + value_2 / 2 + value_3)
+
+            return value
+
+        def compute_integral_restricted_constant(
+                q_1, q_2, x_min, x_max, y_min, y_max):
```

Comment:
What is the difference between this function and the previous one, and do both really need to be implemented here ? If there is one that is better than the other, the other should be removed.